flink-runtime/src/main/java/org/apache/flink/runtime/highavailability/ClientHighAvailabilityServices.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.highavailability;
+
+import org.apache.flink.runtime.leaderretrieval.LeaderRetrievalService;
+
+/**
+ * {@code ClientHighAvailabilityServices} provides services those are required
+ * in client-side. At the moment only web monitor leader retriever is required
+ * because all requests from client are received and propagated by web monitor.
+ */
+public interface ClientHighAvailabilityServices extends AutoCloseable {
+
+	/**
+	 * Get the leader retriever for the web monitor.
+	 *
+	 * @return the leader retriever for the web monitor.
+	 */
+	LeaderRetrievalService getWebMonitorLeaderRetriever();
```

Comment:
I think we now use `RestEndpoint` & `WebMonitor` interchangeably. I am not sure which one should be used. Could you provide more background here Till?

Reply:
Technically speaking it should be the `RestEndpoint` because `WebMonitor` denotes for me more the web UI. It just happens to be the case that our `WebMonitor` runs on the same netty server as the cluster `RestEndpoint`. Hence I would be in favour of calling it `getRestEndpointLeaderRetriever` or `getClusterRestEndpointLeaderRetriever`.

Reply:
Now I'm ok to keep `RestEndpoint` when we refer to rest requests handler. For `WebMonitor` I regard it as a concept that only refers to Web UI(flink-runtime-web). Updated names.

Well, also I have once meet the case that we couple netty channel started and leader contender started which introduce unnecessary dependency(mainly caused by we inherit WebMonitor from RestServerEndpoint). I will open an issue to discuss it once I make that case clear.

Reply:
I'd prefer `getClusterRestEndpointLeaderRetriever`. Updated.